### PR TITLE
Refactors/rewrites most of implant code

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

All of this was bay or cm code

## Why It's Good For The Game

Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when Shotgun arm when 

## Changelog
:cl:
refactor: Refactors/rewrites most of implant code
/:cl:

### CODE REVIEW

**Comment by Kuro-ai** [comment]
when you code it hue hue hue hue hue hue hue hue

File: code/game/objects/items/implants/implant.dm [hunk 1]
```diff
@@ -1,66 +1,114 @@
...
+	var/implanted = FALSE
+	///Owner mob this implant is inserted to
+	var/mob/living/implant_owner
+	///The owner limb this implant is inside
+	var/datum/limb/part
+	///Color of the implant, used for selecting the implant cases icon_state
+	var/implant_color = "b"
+	///whether this implant allows reagents to be inserted into it
+	var/allow_reagents = FALSE
+	///What level of mmalfunction/breakage this implant is at, used for functionality checks
```
Comment (SplinterGP) [inline code/game/objects/items/implants/implant.dm:15]: small typo

File: code/game/objects/items/implants/implantneurostim.dm [hunk 2]
```diff
@@ -1,112 +1,72 @@
...
+	user.mind.store_memory("[src] in [target] can be made to deliver negative stimulus by saying something containing the phrase ''[phrase]'', <B>say [phrase]</B> to attempt to activate.", 0, 0)
+	to_chat(user, "<span class='notice'>[src] in [target] can be made to deliver negative stimulus by saying something containing the phrase ''[phrase]'', <B>say [phrase]</B> to attempt to activate.</span>")
+	return TRUE
 
 
 /obj/item/implant/neurostim/emp_act(severity)
-	if (malfunction)
+	if(malfunction)
 		return
 	if (prob(80))
```
Comment (SplinterGP) [inline code/game/objects/items/implants/implantneurostim.dm:69]: Prob(80) hauehwiqekjwekqejqh, just memeing.

File: code/game/objects/items/implants/implant.dm (hunk 1 above)
Comment (SplinterGP) [reply code/game/objects/items/implants/implant.dm:15]:
```suggestion
	///What level of malfunction/breakage this implant is at, used for functionality checks
```

**Pull request merged**